c++/src/kj/compat/http.h

```diff
@@ -660,10 +661,15 @@ class WebSocket {
   // an empty string indicates a preference for no extensions to be applied.
 };
 
+using TlsStarterCallback = kj::Maybe<kj::Function<kj::Own<kj::AsyncIoStream>(kj::StringPtr)>>;
 struct HttpConnectSettings {
   bool useTls = false;
   // Requests to automatically establish a TLS session over the connection. The remote party
   // will be expected to present a valid certificate matching the requested hostname.
+  kj::Maybe<TlsStarterCallback&> tlsStarter;
+  // An optional reference to a callback. If it's non-null then it may get filled with a
+  // callback function. This callback function can then be used to initiate a TLS handshake on the
+  // connection. The returned AsyncIoStream is a secure TLS stream.
```

Comment:
I'd suggest a couple clarifications:

1. This is an output parameter. Naming it something like `tlsStarterOutputParam` would be ugly, but make it clearer what's going on.
2. It may be worth mentioning that if `tlsStarter` is non-null, but the referent is null after `connect()` is called, that means libkj was compiled without TLS support.
3. What happens if `useTls = true`, and `tlsStarter` is non-null? Is that something we should allow?

Reply:
> What happens if useTls = true, and tlsStarter is non-null? Is that something we should allow?

No, in this case the tlsStarter will not be filled. I'll mention this in the comment as well.